tslib/src/functions/check-marker/check-marker.v1.ts

```diff
@@ -28,17 +28,19 @@ export class CheckMarkerV1 {
   }
 
   async mark (functionBindings: ICheckMarkerFunctionBindings, logger: ILogger): Promise<void> {
-
+    logger.verbose('mark() called')
     const validatedCheck = this.findValidatedCheck(functionBindings.receivedCheckTable)
     const markingData = await this.validateData(functionBindings, validatedCheck, logger)
```

Comment:
This method can return a different response other than undefined or the expected structure of markingData.
The implementation returns the result of `updateReceivedCheckWithMarkingError` in many cases which seems to be a problem. This will need to be addressed.

Reply:
@GeorgeChatzigiannis you are missing the context of working in Typescript - the strict typing forces you to declare all possible return types (using a pipe) and you cannot write code to check for undefined if you have not specified it as one of those return types.

Reply:
@GuyHarwood I can see `updateReceivedCheckWithMarkingError` can return the result of `tableService.replaceEntityAsync` which is being referenced in the `IAsyncTableService` interface.
In there the return type is described as `Promise<any>`.
Am I missing something else in here? 🤔

Reply:
The Promise<any> returns type was masking the issue.  I've updated the types on storage helper.